### PR TITLE
First pass

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
First pass for minimal did-whisper cli client.

* [x] Convert ed25519 (from a DID document) signing key to a Curve25519 encryption key
* [x] Tests to encrypt/decrypt message
* [x] Use a modified `did-client` version to fetch a DID document from a test net
* [x] Wire together with a basic cli harness (that uses stdin/stdout to deal with the messages)
* [x] Create a simple Express API to store/retrieve messages, generate a minified URL

### CODE REVIEW

**Comment by mattcollier** [comment]
@dmitrizagidulin wrt commit messages: https://github.com/digitalbazaar/bedrock/blob/master/CONTRIBUTING.md#commit-messages

**Comment by dmitrizagidulin** [comment]
@mattcollier thanks, will fix.

File: lib/index.js
Comment (mattcollier) [inline lib/index.js:221]: the latest convention we're using is to do destructuring assignment in the argument like:
```
// in alphabetical order
function getDidDoc({client, did, location, mode}) {
```
which is nice because it's somewhat self-documenting, and allows to easily set defaults if needed.  We did the more opaque `options` thing for a long time, and I feel that the new way is much better.

Comment (dmitrizagidulin) [reply lib/index.js:221]: Ah, nice! I fix. :)

Comment (dmitrizagidulin) [reply lib/index.js:221]: My only hesitation is - those properties documented are just pass-through, they’re not really used in the function. But it may be worth for consuming code to know about them. Maybe it doesn’t quite make sense to destructure them here?

Comment (dlongley) [reply lib/index.js:221]: Can do `{foo, bar, ...options}` I believe for the pass through stuff.

File: bin/did-whisper
Comment (mattcollier) [inline bin/did-whisper]: I'm thinking that the default should be https://whisper.demo.veres.one

Which would allow someone to most easily try this thing out without being concerned at all about specifying a storage location.  Whereas now, the default expects one to be running the whisper-server locally.

It does seem like a good idea to guide devs to the proper settings for dev mode as well.  So I don't know, maybe a --dev option that sets the default port to what it is now unless it's overridden by the -s option?

Comment (dmitrizagidulin) [reply bin/did-whisper]: Good idea, on both counts.

**Comment by dmitrizagidulin** [comment]
Demo'd, ready to merge to master.

File: bin/did-whisper
Comment (mattcollier) [inline bin/did-whisper]: No need for {} and return here.

File: lib/index.js
Comment (mattcollier) [inline lib/index.js]: I don't think it's necessary to change all these APIs, but just a reminder that destructured options are preferred over an ordered list of arguments.

File: lib/index.js
Comment (mattcollier) [inline lib/index.js]: Would it be good to have some indication that `cipher` is a buffer, or would it make sense to just do:
```
const cipher = bs58.encode(encryptMessage(message, cipherKeys));
```
or should `encryptMessage` allow for specifying a return type?  Which maybe defaults to b58?

File: lib/index.js
Comment (mattcollier) [inline lib/index.js]: One line?

File: lib/index.js
Comment (mattcollier) [inline lib/index.js]: I am personally averse to mutating arguments, but others are less sensitive.

File: lib/index.js
Comment (mattcollier) [inline lib/index.js]: I like the `!(a && b)` pattern here.

File: lib/index.js
Comment (mattcollier) [inline lib/index.js]: to my earlier point about returning buffers vs strings.  Should probably be consistent in some way.

File: lib/index.js
Comment (mattcollier) [inline lib/index.js:128]: Is throwing here desirable vs some error handling?

File: lib/index.js
Comment (mattcollier) [inline lib/index.js]: one line functions like this are not in DB style.

File: lib/index.js
Comment (mattcollier) [inline lib/index.js]: generally, several objects that could be a one liner.

**Review by mattcollier — APPROVED** [review]
Just a bunch of style nits...  not necessary to make any changes.

File: lib/index.js
Comment (dlongley) [reply lib/index.js]: Yeah, I don't have an issue with it. I believe functions should be short enough that it shouldn't matter. This becomes trickier when we have huge `async.auto` functions -- which I'd like to remove over time anyway with async+await.

**Review by dlongley — APPROVED** [review]
LGTM -- I agree with @mattcollier's comments, though I think it would be better to take care of most them before we merge to set a good baseline for this lib.

File: README.md
Comment (davidlehn) [inline README.md]: Remove this line before merging.

File: README.md
Comment (davidlehn) [inline README.md]: Should add LICENSE file and update this.

Comment (davidlehn) [reply README.md]: Use BSD 3 clause license.

File: lib/index.js
Comment (davidlehn) [inline lib/index.js]: Already using async/await elsewhere, could use it here too.

File: lib/index.js
Comment (davidlehn) [inline lib/index.js]: Could use async/await here.

File: package.json
Comment (davidlehn) [inline package.json]: use 8.6

File: lib/index.js
Comment (davidlehn) [inline lib/index.js:225]: For a demo this is fine for now, but will want to use some other API like did-io soon.  That API was not meant to be called like this.  I think it could call the callback many times with certain options.

File: bin/did-whisper
Comment (dmitrizagidulin) [reply bin/did-whisper]: I generally use the short form (with no return or {}) for promises only when it fits on one line. The ternary operator here doesn't fit on one line, and looks weird without the curly braces.

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js]: Got it, fixing..

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js]: Thanks, changed.

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js]: Agree in general, though in this particular case I think it reads clearer using `||`. Will change, though, since you both request.

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js:128]: I think I meant it as a reminder that any sort of network operation might result in an error. My intention was that those should be handled by the caller code, rather than here, but am open to suggestions.

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js]: Thanks, fixing.

File: package.json
Comment (dmitrizagidulin) [reply package.json]: Fixed.

File: README.md
Comment (dmitrizagidulin) [reply README.md]: Updated.

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js]: Fixed.

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js:225]: Yeah, agreed, will be changed as soon as did-io refactors are done.

File: README.md
Comment (dmitrizagidulin) [reply README.md]: Done.

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js]: Thx, fixed.

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js]: Thanks. Went with `const cipher = bs58.encode(encryptMessage(message, cipherKeys));`.

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js]: Fixed.

File: lib/index.js
Comment (dmitrizagidulin) [reply lib/index.js]: Makes sense, changed to return buffer, for consistency.

**Comment by dmitrizagidulin** [comment]
@mattcollier @davidlehn @dlongley - Made changes to reflect your feedback. Let me know if there's anything else.

**Comment by msporny** [comment]
Are we good with this PR @mattcollier and @dlongley? I'm waiting on it to land to merge https://github.com/veres-one/docs.veres.one/pull/8

**Pull request merged**
